### ISSUE TITLE
`object-shorthand` not working properly with Object Rest Properties

### ISSUE THREAD

**Issue by eTorAken** [eslint/eslint#2820]
I'm using ESLint `0.23.0` and it seems to have a bug on the `object-shorthand` rule using [Object Rest Properties](https://github.com/sebmarkbage/ecmascript-rest-spread#rest-properties). For instance, this code:
```js
let { x, y, ...z } = { x: 1, y: 2, a: 3, b: 4 };
x; // 1
y; // 2
z; // { a: 3, b: 4 }
```
is throwing `Expected property shorthand  object-shorthand`.

Moreover, it is also throwing a `Unexpected token` error (which might be related to https://github.com/eslint/eslint/issues/2346).

I activated `ecmaFeatures.restParams` but nothing has changed.

**Comment by lo1tuma** [comment]
Which parser do you use?

The default ESLint parser doesn’t support object rest properties.

**Comment by eTorAken** [comment]
Sorry I did not know, I'm using `babel-eslint` so it should be directly related to it.

**Comment by sebmck** [comment]
See babel/babel-eslint#131.

**Comment by ilyavolodin** [comment]
Closing as this is not an ESLint issue. We do not natively support object rest properties.

**Comment by xjamundx** [comment]
Now that we do support rest object is this a valid issue?

**Comment by JaxCavalera** [comment]
Is this working yet or still broken?
